Docs: jsdoc example using `build.jobs` and `build.tools` (#9241)
readthedocs/readthedocs.org: docs/user/build-customization.rst

         - ./git-lfs fetch
         # Make local files to have the real content on them
         - ./git-lfs checkout

> Document your code with ``jsdoc``
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: This might be better off generalized, rather than specific to jsdoc. There is a lot more to say about jsdoc usage, and duplicating that here will probably guide users to the wrong guide content.

```suggestion
Install Node.js dependencies
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Author: I like this idea, :100:

It's possible to make usage of ``jsdoc`` to document your Javascript code directly from your source code.

> To setup it, you need to first define the version of ``nodejs`` you want to use and also install ``jsdoc`` as shown below:

Reviewer: If generalizing this, the copy here would need to be tweaked. But generally, noting to use the official terminology:

```suggestion
To setup it, you need to first define the version of Node.js you want to use and also install ``jsdoc`` as shown below:
```

.. code:: yaml

   version: 2
   build:
     os: "ubuntu-22.04"
     tools:
       python: "3.9"
       nodejs: "16"
     jobs:
       post_install:

>          - npm install -g jsdoc

Reviewer 2: Seems like this is missing the part where you actually implement it in Sphinx? Could be neat to include to make it more copy-pastable.

Author: Ah, yeah. I wasn't sure if this page is the right place for that. I felt that I was contaminating it with a lot of code that it's outside the scope of `build.jobs` / build customization. Probably, the full example fits better on a blog post or a guide explaining the full process to document Javascript documentation. We could link to that post/guide from here.

Reviewer: Also, if we can generalize this example, maybe it's wise to point out `npm ci` as well. This is probably what the user wants ultimately, for reproducible builds, so would be a good suggestion for us to give.

```suggestion
         - npm ci
         - npm install -g jsdoc
```
